File: submit_story.js
document.getElementById("storyForm").addEventListener("submit", function(event) {
    event.preventDefault();
    
    var formData = new FormData(this);
    var jsonData = {};
    
    formData.forEach(function(value, key) {
        jsonData[key] = value;
    });
    
    fetch("submit_story.py", {
        method: "POST",
        body: JSON.stringify(jsonData),
        headers: {
            "Content-Type": "application/json"
        }
    })
    .then(response => {
        if (!response.ok) {
            throw new Error("Network response was not ok");
        }
        return response.text();
    })
    .then(data => {
        console.log(data); 
        // Redirect to thank you page
        window.location.href = "thank_you.html";
    })
    .catch(error => {
        console.error("There was a problem with your fetch operation:", error);
    });
});

File: submit_story.py
# submit_story.py

import json

# Handle POST request
def handle_post(request):
    # Get the JSON data from the request
    data = json.loads(request.data)
    
    # Here you can process the data as needed, such as storing it in a database
    # For now, let's just print the data
    print(data)
    
    # Respond with a success message
    return "Success"

# Main function to handle requests
def application(environ, start_response):
    status = '200 OK'
    headers = [('Content-type', 'text/plain; charset=utf-8')]
    
    # Check the request method
    if environ['REQUEST_METHOD'] == 'POST':
        # Get the request data
        request_body_size = int(environ.get('CONTENT_LENGTH', 0))
        request_body = environ['wsgi.input'].read(request_body_size)
        
        # Call the function to handle the POST request
        response_body = handle_post(request_body)
    else:
        # If method is not POST, return an error message
        response_body = "Method not allowed"
    
    # Send the response
    start_response(status, headers)
    return [response_body.encode()]
